eventsourced/src/lib.rs

```diff
@@ -288,20 +292,20 @@ impl<E> EventSourcedExt for E where E: EventSourced {}
 /// A handle for a spawned event sourced entity which can be used to invoke its command handler.
 #[derive(Debug, Clone)]
 #[allow(clippy::type_complexity)]
-pub struct EntityRef<E>
+pub struct EntityRef<E, T>
```

Comment:
IIUC, this means that the result type has to be the same for each command. Is the idea that the user of the library then has another `enum` for the responses?

This probably makes it easier to support getting the response back across from the entity but in some way we are shifting the responsibility that responses to a command fit the command to the user.

I'm trying out how we could have each command have its response and check that statically.

Reply:
Statically assigning responses to commands will be equally hard like errors to commands. The core issue are the channels used for messaging btw tasks (handler and entityref), they are typed to the overall error type.